src/main/java/frc/robot/subsystems/Pivot.kt

```diff
@@ -71,25 +78,20 @@ object Pivot : SubsystemBase("Pivot") {
 
     init {
         follower.setControl(Follower(9, true))
-        defaultCommand = Commands.idle(this)
+        //        defaultCommand = run {
+        // leader.setControl(motionMagic.withPosition(currentSetpoint)) }
     }
 
     fun goTo(state: RobotState): Command =
         PrintCommand("Pivot going to $state - ${state.pivotAngle}")
+            // leader.setControl(motionMagic.withPosition(state.pivotAngle)) }))
             .alongWith(runOnce { leader.setControl(motionMagic.withPosition(state.pivotAngle)) })
 
-    //    private val isVertical: BooleanSupplier = BooleanSupplier {Pivot.angle > 80.degrees &&
-    // Pivot.angle < 95.degrees}
-
     fun goToAndWaitUntilVertical(state: RobotState): Command =
-        PrintCommand(
-                "Pivot going to $state - ${state.pivotAngle} from $angle and waiting until vertical"
-            )
-            .alongWith(runOnce { leader.setControl(motionMagic.withPosition(state.pivotAngle)) })
-            .andThen(Commands.idle())
-            .until {
-                (angle - state.pivotAngle < 3.degrees) && (angle - state.pivotAngle > (-3).degrees)
-            }
+        PrintCommand("Pivot going vertical").alongWith(goTo(state)).andThen(Commands.idle()).until {
+            abs(angle.baseUnitMagnitude() - state.pivotAngle.baseUnitMagnitude()) <
+                ELEVATOR_THRESHOLD.baseUnitMagnitude()
```

Comment:
```suggestion
            abs((angle - state.pivotAngle).degrees) <
                ELEVATOR_THRESHOLD.degrees
```
or check if the closedLoopError < Elevator_Threshold